frontend/src/app/shared/models/match.model.ts

```diff
@@ -0,0 +1,19 @@
+import { Roster } from "./roster.model";
+
+export class Match {
+  public id: number;
+  public teamA: Roster;
+  public teamB: Roster;
+  public winner: number | null;
```

Comment:
This was a little unclear to me at first. I'm assuming its for the winning Owner ID, right? Maybe change to winnerID or something. It also feels like the winner field should hold the same data as the team fields. Like if TeamA is using Roster, then winner should also use Roster.